lib/src/trace.dart

```diff
@@ -285,7 +285,10 @@ class Trace implements StackTrace {
         var library = frame.library.replaceAll(_terseRegExp, '');
         return new Frame(Uri.parse(library), null, null, frame.member);
       }).toList();
-      if (newFrames.length > 1 && newFrames.first.isCore) newFrames.removeAt(0);
+
+      if (newFrames.length > 1 && predicate(newFrames.first)) {
```

Comment:
This trimming logic happens after the frames are folded, which means that all frames matching `predicate()` at the beginning of `newFrames` have already been folded into a single frame—in otherwords, it's guaranteed that if `predicate(newFrames.first)`, then `!predicate(newFrames[1])`. So this is stable under repeated iterations.